Added tests for Function::Parameters parser

jimmygoogle opened the pull request:
I added some tests for the Perl Function::Parameters parser

masatake commented on Units/function-parameters.r/c-fp-test.d/expected.tags:
```diff
@@ -0,0 +1,3 @@
+fun	input.c	/^int fun(void) $/;"	f	typeref:typename:int
+method	input.c	/^int method(void) $/;"	f	typeref:typename:int
```
A whitespace after `(void)` is not needed.

masatake commented on Units/function-parameters.r/c-fp-test.d/expected.tags:
```diff
@@ -0,0 +1,3 @@
+fun	input.c	/^int fun(void) $/;"	f	typeref:typename:int
```
A whitespace after `(void)` is not needed.

masatake commented on Units/function-parameters.r/c-fp-test.d/expected.tags:
```diff
@@ -0,0 +1,3 @@
+fun	input.c	/^int fun(void) $/;"	f	typeref:typename:int
+method	input.c	/^int method(void) $/;"	f	typeref:typename:int
+sub	input.c	/^int sub(void) $/;"	f	typeref:typename:int
```
A whitespace after `(void)` is not needed.

masatake commented on Units/function-parameters.r/c-fp-test.d/input.c:
```diff
@@ -0,0 +1,14 @@
+int method(void)
```
...becase there is no whitespace after `(void)`.

masatake commented on Units/function-parameters.r/python-fp-test.d/expected.tags:
```diff
@@ -0,0 +1,3 @@
+fun	input.py	/^def fun():$/;"	f
```
ctags never runs Perl parser ( and FunctionParameter subparser ) for .py files.
So this test case doesn't make sense as a test case for FunctionParameter subparser.
This test case tests the main part of ctags.
Please remove this test case.

masatake reviewed (changes requested):
Thank you.

masatake commented:
Thank you!

jimmygoogle commented:
Thanks for your help with the parser. I wish I could have done it but I
dont know C that well.

On Mon, Feb 15, 2021, 8:10 PM Masatake YAMATO <notifications@github.com>
wrote:

> Thank you!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/universal-ctags/ctags/pull/2868#issuecomment-779520781>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABGY4TI7OG5R26W2KSTBQR3S7HAZDANCNFSM4XTZMEQA>
> .
>

masatake commented:
I got a question. c-fp-test.d/input.c is not for testing FunctionParameters as I wrote python-fp-test.d, isn't it?

jimmygoogle commented:
Well I added tests for C and Python to prove the parser didn't try go
beyond the Perl files.

On Mon, Feb 15, 2021, 8:18 PM Masatake YAMATO <notifications@github.com>
wrote:

> I got a question. c-fp-test.d/input.c is not for testing
> FunctionParameters as I wrote python-fp-test.d, isn't it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/universal-ctags/ctags/pull/2868#issuecomment-779522665>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABGY4TPHQAADTYADVIXTNS3S7HBWFANCNFSM4XTZMEQA>
> .
>

masatake commented:
> Well I added tests for C and Python to prove the parser didn't try go beyond the Perl files.

I deleted them in #2870 there is no code choosing a parser in the FunctionParameters subparser and Perl parser.
The testing cases for choosing a parser for an input file are at Tmain directory.

jimmygoogle commented:
OK I see, I had just added to make sure the parser didnt pick them up.

On Mon, Feb 15, 2021 at 8:28 PM Masatake YAMATO <notifications@github.com>
wrote:

> Well I added tests for C and Python to prove the parser didn't try go
> beyond the Perl files.
>
> I deleted them in #2870
> <https://github.com/universal-ctags/ctags/pull/2870> there is no code
> choosing a parser in the FunctionParameters subparser and Perl parser.
> The testing cases for choosing a parser for an input file are at Tmain
> directory.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/universal-ctags/ctags/pull/2868#issuecomment-779525516>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABGY4TNKQHY6V76EARIGN4DS7HC3ZANCNFSM4XTZMEQA>
> .
>
